Sources/NIO/LinuxUring.swift

```diff
@@ -0,0 +1,509 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if SWIFTNIO_LIBURING
+#if os(Linux)
+
+import CNIOLinux
+
+// we stuff the event type into the user data for the sqe together with
+// the fd and sequenceIdentifier to match the events without needing any
+// memory allocations or references.
+
+internal enum CqeEventType : Int {
+    case poll = 1, pollModify, pollDelete // start with 1 to not get zero bit patterns for stdin
```

Comment:
is the 0 bit pattern a problem?

Reply:
It was just to be able to have an assert for zero bit pattern (to sanity check), there was naturally a zero bit pattern when running with stdin on one of the integration tests. It could be put to zero and remove the assert now.

Reply:
(although easier to read the debug output, so even while the assert is now removed, keeping event type starting from 1 for easier reading of debug output)